pkg/vsphere/simulator/simulator_test.go

```diff
@@ -266,6 +271,74 @@ func TestServeHTTP(t *testing.T) {
 	}
 }
 
+func TestServeHTTPS(t *testing.T) {
+	s := New(NewServiceInstance(esx.ServiceContent, esx.RootFolder))
+	s.TLS = new(tls.Config)
+	ts := s.NewServer()
+	defer ts.Close()
+
+	ctx := context.Background()
+	// insecure=true OK
+	client, err := govmomi.NewClient(ctx, ts.URL, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = client.Login(ctx, ts.URL.User)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// insecure=false should FAIL
+	_, err = govmomi.NewClient(ctx, ts.URL, false)
+	if err == nil {
+		t.Fatal("expected error")
+	}
+
+	uerr, ok := err.(*url.Error)
+	if !ok {
+		t.Fatalf("err type=%T", err)
+	}
+
+	_, ok = uerr.Err.(x509.UnknownAuthorityError)
+	if !ok {
+		t.Fatalf("err type=%T", uerr.Err)
+	}
+
+	sinfo := ts.CertificateInfo()
+
+	// Test custom verify
+	sc := soap.NewClient(ts.URL, false)
+	sc.SetDialTLS(func(err error, state tls.ConnectionState) error {
+		pinfo := new(object.HostCertificateInfo).FromCertificate(state.PeerCertificates[0])
+		// peer thumbprint should match our server thumbprint
+		if pinfo.ThumbprintSHA1 != sinfo.ThumbprintSHA1 {
+			return err
+		}
+		return nil
+	})
+
+	_, err = vim25.NewClient(ctx, sc)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Test custom RootCAs list
+	sc = soap.NewClient(ts.URL, false)
+	caFile, err := ts.CertificateFile()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err = sc.SetRootCAs(caFile); err != nil {
```

Comment:
No objection to this, but wondering if it'd be easier just to pass around the certificate directly instead of dealing with tmp files.